app/views/question_sets/_form.html.erb

```diff
@@ -26,7 +26,62 @@
     <% end %>
   </div>
 
+  <!-- check if current_user's courses have any questions
+        and allow user to filter those questions by searching tags.-->
+  <% if @questions.present? %>
+    <!-- add "search bar" for tags and filter questions shown based on
+          tag entered -->
+
+    <input type="text" id="tagSearchInput" onkeyup="qSetQTagSearch()" placeholder="Search for question tags...">
+
+    <table id="questionTagTable" border="1">
+      <tr>
+        <th></th>
+        <th style="min-width:250px">Body</th>
+        <th style="min-width:100px">Tags</th>
+      </tr>
+      <!-- loop thru questions from controller
+            (questions associated with courses the current_user is
+            a professor or assistant for) -->
+      <% @questions.each do |q| %>
```

Comment:
Almost forgot one last thing, I think this table is supposed to replace this: https://github.com/jds2017/ALRails/blob/59a0873f52b9e90e0f64126fb727a74b0e2ac04e/app/views/question_sets/_form.html.erb#L19

I think you can delete that^ but then you'll also need to copy the logic for the checkbox. I also noticed that when I tried to save something with nothing checked, it errored out